File: C++/Maths/SumOfNumbers.cpp
/*
Explanation of Approach:
- This program first takes an integer input `n`, which represents the number of elements the user wants to sum.
- It then enters a loop that runs `n` times, where in each iteration, it takes an integer input `num` from the user and adds it to the variable `sum`.
- After the loop finishes, the program outputs the calculated sum using `std::cout`.

Time Complexity:
- The time complexity of this program is O(n), where `n` is the number of elements to sum. The loop runs `n` times, and each iteration takes a constant amount of time.

Space Complexity:
- The space complexity of this program is O(1) because it uses only a constant amount of additional memory to store the variables `n`, `sum`, and `num`.

Sample Input:
Enter the number of elements: 5
Enter element 1: 10
Enter element 2: 20
Enter element 3: 30
Enter element 4: 40
Enter element 5: 50

Sample Output:
Sum: 150

In the sample input, the user entered 5 elements (10, 20, 30, 40, and 50), and the program calculated their sum, which is 150, and displayed the result as output.
*/

#include <iostream>

int main() {
    int n;
    int sum = 0;

    // Input the number of elements
    std::cout << "Enter the number of elements: ";
    std::cin >> n;

    // Input the elements and calculate the sum
    for (int i = 0; i < n; ++i) {
        int num;
        std::cout << "Enter element " << i + 1 << ": ";
        std::cin >> num;
        sum += num;
    }

    // Output the sum
    std::cout << "Sum: " << sum << std::endl;

    return 0;
}

File: Python/Maths/SumOfNumbers.py
'''
Explanation of Approach:
- The program defines a function `sum_of_numbers` that takes a list of numbers as input.
- Inside the function, it initializes a variable `total_sum` to 0, which will store the sum of the numbers.
- It then iterates through each number in the input list using a `for` loop and adds each number to the `total_sum`.
- After processing all the numbers, the function returns the final `total_sum`.
- In the main part of the program, we provide a sample input list `numbers_list` containing `[2, 4, 6, 8, 10]`.
- We call the `sum_of_numbers` function with this input list and store the result in `result_sum`.
- Finally, the program outputs the result as "Sum: 30".

Time Complexity:
- The time complexity of this program is O(N), where N is the number of elements in the input list. The `for` loop iterates through each element once, and the time taken is directly proportional to the size of the input list.

Space Complexity:
- The space complexity of this program is O(1) because it only uses a constant amount of additional memory, regardless of the size of the input list. The memory usage remains constant as the program only uses a single variable (`total_sum`) to store the result.

Sample Input:
numbers_list = [2, 4, 6, 8, 10]

Sample Output:
Sum: 30

In the sample input, the function `sum_of_numbers` calculates the sum of the numbers in the list `[2, 4, 6, 8, 10]`, and the output is `Sum: 30`.
'''

def sum_of_numbers(numbers):
    """
    Function to calculate the sum of numbers in a list.

    Parameters:
    numbers (list): A list of numbers.

    Returns:
    int: The sum of the numbers in the list.
    """
    total_sum = 0

    # Iterate through each number in the list and add it to the total_sum
    for num in numbers:
        total_sum += num

    return total_sum

# Sample Input
numbers_list = [2, 4, 6, 8, 10]

# Calculate the sum of numbers in the list
result_sum = sum_of_numbers(numbers_list)

# Sample Output
print("Sum:", result_sum)

